http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/BaselineDependenciesBuilder.scala

```diff
@@ -0,0 +1,468 @@
+      cbasDao,
+      cromwellDao,
+      hailBatchDao,
+      listenerDao,
+      wsmClientProvider,
+      azureContainerService,
+      runtimeServiceConfig,
+      kubernetesDnsCache,
+      appDescriptorDAO,
+      helmClient,
+      azureRelay,
+      azureVmService,
+      operationFutureCache,
+      azureBatchService,
+      azureApplicationInsightsService,
+      openTelemetry
+    )
+
+  private def createCloudSubscriber[F[_]: Parallel](
+    subscriberQueue: Queue[F, ReceivedMessage[LeoPubsubMessage]]
+  )(implicit F: Async[F], logger: StructuredLogger[F]): Resource[F, CloudSubscriber[F, LeoPubsubMessage]] =
+    ConfigReader.appConfig.azure.hostingModeConfig.enabled match {
+      case false =>
+        GoogleSubscriber.resource[F, LeoPubsubMessage](subscriberConfig, subscriberQueue)
+      case true =>
+        AzureSubscriberInterpreter.subscriber[F, LeoPubsubMessage](
+          ConfigReader.appConfig.azure.hostingModeConfig.subscriberConfig,
+          subscriberQueue
+        )
+    }
+
+  private def createCloudPublisher[F[_]](implicit
+    F: Async[F],
+    logger: StructuredLogger[F]
+  ): Resource[F, CloudPublisher[F]] =
+    ConfigReader.appConfig.azure.hostingModeConfig.enabled match {
+      case false =>
+        GooglePublisher.cloudPublisherResource[F](publisherConfig)
+      case true =>
+        AzurePublisherInterpreter.publisher[F](ConfigReader.appConfig.azure.hostingModeConfig.publisherConfig)
+    }
+
+  private def buildCache[K, V](maxSize: Int,
+                               expiresIn: FiniteDuration
+  ): com.github.benmanes.caffeine.cache.Cache[K, V] =
+    Caffeine
+      .newBuilder()
+      .maximumSize(maxSize)
+      .expireAfterWrite(expiresIn.toSeconds, TimeUnit.SECONDS)
+      .recordStats()
+      .build[K, V]()
+
+  private def buildHttpClient[F[_]: Async: StructuredLogger](
+    sslContext: SSLContext,
+    dnsResolver: RequestKey => Either[Throwable, InetSocketAddress],
+    metricsPrefix: Option[String],
+    withRetry: Boolean
+  ): Resource[F, org.http4s.client.Client[F]] = {
+    // Retry all SocketExceptions to deal with pooled HTTP connections getting closed.
+    // See https://broadworkbench.atlassian.net/browse/IA-4069.
+    val retryPolicy = RetryPolicy[F](
+      RetryPolicy.exponentialBackoff(30 seconds, 5),
+      (req, result) =>
+        result match {
+          case Left(e) if e.isInstanceOf[SocketException] => true
+          case _                                          => RetryPolicy.defaultRetriable(req, result)
+        }
+    )
+
+    for {
+      httpClient <- client
+        .BlazeClientBuilder[F]
+        .withSslContext(sslContext)
+        // Note a custom resolver is needed for making requests through the Leo proxy
+        // (for example HttpJupyterDAO). Otherwise the proxyResolver falls back to default
+        // hostname resolution, so it's okay to use for all clients.
+        .withCustomDnsResolver(dnsResolver)
+        .withConnectTimeout(30 seconds)
```

Comment:
is it worth having this live in `reference.conf`? Might be worth considering that or adding to be configurable based on env in `leo.conf`

Reply:
This is how it is in the current Boot file, and I agree that a conf value makes sense. However, it's out scope for this ticket, but a valid follow up item.